app/src/features/datasets/components/security/ApplySecurityDialog.tsx

```diff
@@ -0,0 +1,176 @@
+import { useTheme } from '@mui/material/styles';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import YesNoDialog from 'components/dialog/YesNoDialog';
+import { DialogContext, ISnackbarProps } from 'contexts/dialogContext';
+import { Formik, FormikProps } from 'formik';
+import { useApi } from 'hooks/useApi';
+import { IArtifact } from 'interfaces/useDatasetApi.interface';
+import React, { useContext, useRef, useState } from 'react';
+import { ISecurityReason } from './SecurityReasonCategory';
+import SecurityReasonSelector, {
+  SecurityReasonsInitialValues,
+  SecurityReasonsYupSchema
+} from './SecurityReasonSelector';
+import SelectedDocumentsDataset from './SelectedDocumentsDataset';
+export interface IApplySecurityDialog {
+  selectedArtifacts: IArtifact[];
+  open: boolean;
+  onClose: () => void;
+}
+
+/**
+ * Publish button.
+ *
+ * @return {*}
+ */
+const ApplySecurityDialog: React.FC<IApplySecurityDialog> = (props) => {
+  const biohubApi = useApi();
+  const { selectedArtifacts, open, onClose } = props;
+
+  const theme = useTheme();
+  const fullScreen = useMediaQuery(theme.breakpoints.down('xl'));
+
+  const dialogContext = useContext(DialogContext);
+  const [yesNoDialogOpen, setYesNoDialogOpen] = useState(false);
+
+  const showSnackBar = (textDialogProps?: Partial<ISnackbarProps>) => {
+    dialogContext.setSnackbar({ ...textDialogProps, open: true });
+  };
+
+  const [formikRef] = useState(useRef<FormikProps<any>>(null));
+
+  const handleSubmit = async (securityReasons: ISecurityReason[]) => {
+    await biohubApi.security.applySecurityReasonsToArtifacts(selectedArtifacts, securityReasons);
+  };
+
+  return (
+    <>
+      <Dialog
+        fullScreen={fullScreen}
+        maxWidth="xl"
+        open={open}
+        aria-labelledby="component-dialog-title"
+        aria-describedby="component-dialog-description"
+        PaperProps={{
+          sx: {
+            width: '100%',
+            height: '100%'
+          }
+        }}>
+        <Formik
+          innerRef={formikRef}
+          initialValues={SecurityReasonsInitialValues}
+          validationSchema={SecurityReasonsYupSchema}
+          validateOnBlur={true}
+          validateOnChange={false}
+          onSubmit={async (values: { securityReasons: ISecurityReason[] }) => {
+            await handleSubmit(values.securityReasons);
+            showSnackBar({
+              snackbarMessage: (
+                <>
+                  <Typography variant="body2" component="div">
+                    You successfully applied security reasons to the file{selectedArtifacts.length > 1 && 's'}.
+                  </Typography>
+                </>
+              ),
+              open: true
+            });
+            onClose();
+          }}>
+          {(formikProps) => (
+            <>
+              <YesNoDialog
+                open={yesNoDialogOpen}
+                onClose={() => setYesNoDialogOpen(false)}
+                onYes={async () => {
```

Comment:
Could avoid some code duplication by moving all the `showSnackbar` and associated React code into a callback, e.g.:

```tsx
const handleSuccess = () => {
  showSnackBar({
    snackbarMessage: (
      <Typography variant="body2" component="div">
        You successfully unsecured the file{selectedArtifacts.length !== 1 && 's'}.
      </Typography>
    ),
    open: true
  });
}
```

Reply:
Will absorb into next ticket SIMSBIOHUB-108, Continuation of development for this feature